### PR TITLE
Add putBatch method

### PR DESCRIPTION
**Pull request by aheinz-sg**

### CODE REVIEW

File: Tests/BeanstalkdQueueImplementationTest.php [hunk 1]
```diff
@@ -14,6 +14,21 @@ public function tearDown()
     }
 
     /**
+     * @test
+     */
+    public function putBatch()
+    {
+        $jobs = array(1, 2);
```
Comment (wowo) [inline Tests/BeanstalkdQueueImplementationTest.php]: can u please use short notation [] ?

File: Tests/SQSQueueImplementationTest.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    /**
+     * @test
+     * @dataProvider data_putBatch
+     */
+    public function putBatch($jobs, $batchCount)
+    {
+        $tube = 'tube';
+        $queueName = 'queueName';
+
+        $q = $this->getMockBuilder('\Aws\Sqs\SqsClient')->disableOriginalConstructor()->setMethods(array('sendMessage', 'sendMessageBatch', 'getQueueUrl'))->getMock();
```
Comment (wowo) [inline Tests/SQSQueueImplementationTest.php]: short array notation please

File: Tests/SQSQueueImplementationTest.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        $q->expects($this->exactly($batchCount))->method('sendMessageBatch');
+        $q->expects($this->once())->method('getQueueUrl')->with(['QueueName' => $queueName])->will($this->returnValue(new Model(['QueueUrl' => 'queueUrl'])));
+
+        $impl = new SQSQueueImplementation($q, array($tube => $queueName));
+        $impl->putBatch($tube, $jobs);
+    }
+
+    public function data_putBatch()
+    {
+        return array(
```
Comment (wowo) [inline Tests/SQSQueueImplementationTest.php]: short array notation as well pls

File: Tests/SQSQueueImplementationTest.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        $q = $this->getMockBuilder('\Aws\Sqs\SqsClient')->disableOriginalConstructor()->setMethods(array('sendMessage', 'sendMessageBatch', 'getQueueUrl'))->getMock();
+        $q->expects($this->never())->method('sendMessage');
+        $q->expects($this->exactly($batchCount))->method('sendMessageBatch');
+        $q->expects($this->once())->method('getQueueUrl')->with(['QueueName' => $queueName])->will($this->returnValue(new Model(['QueueUrl' => 'queueUrl'])));
+
+        $impl = new SQSQueueImplementation($q, array($tube => $queueName));
+        $impl->putBatch($tube, $jobs);
+    }
+
+    public function data_putBatch()
```
Comment (wowo) [inline Tests/SQSQueueImplementationTest.php]: function name shouldn't have _

File: Tests/SQSQueueImplementationTest.php (hunk 2 above)
Comment (wowo) [inline Tests/SQSQueueImplementationTest.php]: btw. we're using mockery here, wouldn't it be a problem to levarage it here as well?

Comment (aheinz-sg) [inline Tests/SQSQueueImplementationTest.php]: I tried that at first, but Mockery didn't like a final method in \Aws\Sqs\SqsClient, nor when I tried the partial mock with an instantiated class. I'll give it another look when I change the array notation.

**Pull request merged**